scylla/src/transport/session_builder.rs

```diff
@@ -762,6 +762,27 @@ impl<K: SessionBuilderKind> GenericSessionBuilder<K> {
         self
     }
 
+    /// Changes DNS hostname resolution timeout.
+    /// The default is 5 seconds.
+    ///
+    /// # Example
+    /// ```
+    /// # use scylla::{Session, SessionBuilder};
+    /// # use std::time::Duration;
+    /// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// let session: Session = SessionBuilder::new()
+    ///     .known_node("127.0.0.1:9042")
+    ///     .hostname_resolution_timeout(Duration::from_secs(10))
+    ///     .build() // Turns SessionBuilder into Session
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn hostname_resolution_timeout(mut self, duration: Duration) -> Self {
+        self.config.hostname_resolution_timeout = Some(duration);
+        self
+    }
+
```

Comment:
I believe this should accept `Option<Duration>`, especially that the default is `Some`. Now it's impossible to set no timeout with `SessionBuilder`'s API.

Reply:
I decided to be consistent with other optional timeout options. See for example `SessionBuilder::keepalive_interval()` It accepts `Duration`, and sets the corresponding option to `Some(duration)` as well.

If user wants to set any of these to `None`, it can do so by direct access to the field (fields are public):
```rust
sb.config.hostname_resolution_timeout = None;
```

Reply:
Hmm, I dislike the idea of `SessionBuilder` exposing only subset of possible configuration options using methods, with some being available only through fields.
WDYT @Lorak-mmk ?

Reply:
We could expose another methods (`disable_hostname_resolution_timeout`, `disable_keepalive`) which would be a bit more descriptive.